File: TSPGreedy.java
// Java program for the above approach
import java.util.*;
 
public class TSPGreedy {
 
    // Function to find the minimum
    // cost path for all the paths
    static void findMinRoute(int[][] tsp)
    {
        int sum = 0;
        int counter = 0;
        int j = 0, i = 0;
        int min = Integer.MAX_VALUE;
        List<Integer> visitedRouteList
            = new ArrayList<>();
 
        // Starting from the 0th indexed
        // city i.e., the first city
        visitedRouteList.add(0);
        int[] route = new int[tsp.length];
 
        // Traverse the adjacency
        // matrix tsp[][]
        while (i < tsp.length
               && j < tsp[i].length) {
 
            // Corner of the Matrix
            if (counter >= tsp[i].length - 1) {
                break;
            }
 
            // If this path is unvisited then
            // and if the cost is less then
            // update the cost
            if (j != i
                && !(visitedRouteList.contains(j))) {
                if (tsp[i][j] < min) {
                    min = tsp[i][j];
                    route[counter] = j + 1;
                }
            }
            j++;
 
            // Check all paths from the
            // ith indexed city
            if (j == tsp[i].length) {
                sum += min;
                min = Integer.MAX_VALUE;
                visitedRouteList.add(route[counter] - 1);
                j = 0;
                i = route[counter] - 1;
                System.out.println("route[counter]: " + route[counter]);
                System.out.println("i: "+ i);
                counter++;
            }
        }
 
        // Update the ending city in array
        // from city which was last visited
        i = route[counter - 1] - 1;
 
        for (j = 0; j < tsp.length; j++) {
 
            if ((i != j) && tsp[i][j] < min) {
                min = tsp[i][j];
                route[counter] = j + 1;
            }
        }
        sum += min;
 
        // Started from the node where
        // we finished as well.
        System.out.print("Minimum Cost is : ");
        System.out.println(sum);
    }
 
    // Driver Code
    public static void
        main(String[] args)
    {
        // Input Matrix
        int[][] tsp = {
            { -1, 10, 15, 20 },
            { 10, -1, 35, 25 },
            { 15, 35, -1, 30 },
            { 20, 25, 30, -1 }
        };
 
        // Function Call
        findMinRoute(tsp);
    }
}

File: TSPGreedy.js

// https://www.geeksforgeeks.org/travelling-salesman-problem-greedy-approach/

class TSPGreedy {
 
    constructor() {
        
    }
    // Function to find the minimum
    // cost path for all the paths
    findMinRoute(tsp)
    {
        console.log(tsp);
        var sum = 0;
        var counter = 0;
        var j = 0, i = 0;
        var min = Number.NEGITIVE_INFINITY;
        var visitedRouteList = [];
 
        // Starting from the 0th indexed
        // city i.e., the first city
        visitedRouteList.push(0);
        var route = [];
 
        // Traverse the adjacency
        // matrix tsp[][]
        while (i < tsp.length && j < tsp[i].length) {
            // Corner of the Matrix
            if (counter >= tsp[i].length - 1) {
                break;
            }
            // If this path is unvisited then
            // and if the cost is less then
            // update the cost
            if (j != i
                && !(j in visitedRouteList)) {
                if (tsp[i][j] < min) {
                    min = tsp[i][j];
                    route[counter] = j + 1;
                    console.log("route: " + route);
                }
            }
            j++;
            // Check all paths from the
            // ith indexed city
            if (j == tsp[i].length) {
                sum += min;
                min = Number.POSITIVE_INFINITY;
                visitedRouteList.push(route[counter] - 1);
                j = 0;
                i = route[counter] - 1;
                console.log("counter: " + counter);
                console.log("route[counter]: " + route[counter]);
                console.log("i: "+ i);
                counter++;
            }
        }
        // Update the ending city in array
        // from city which was last visited
        i = route[counter - 1] - 1;
        for (var j = 0; j < tsp.length; j++) {
            console.log("i: "+ i);
            console.log("j: "+ j);
            if ((i != j) && tsp[i][j] < min) {
                min = tsp[i][j];
                route[counter] = j + 1;
                console.log("route: " + route);
            }
        }
        sum += min;
        // Started from the node where
        // we finished as well.
        console.log("Minimum Cost is : ");
        console.log(sum);
    }
}

// Input Matrix
var tsp = [
[ -1, 10, 15, 20 ],
[ 10, -1, 35, 25 ],
[ 15, 35, -1, 30 ],
[ 20, 25, 30, -1 ]
];

var tsp_solver = new TSPGreedy();
// Function Call
tsp_solver.findMinRoute(tsp);